internal/k6runner/k6runner.go

```diff
@@ -328,29 +344,98 @@ type RunResponse struct {
 }
 
 func (r HttpRunner) WithLogger(logger *zerolog.Logger) Runner {
-	return HttpRunner{
-		url:    r.url,
-		logger: logger,
-	}
+	r.logger = logger
+
+	return r
 }
 
 var ErrUnexpectedStatus = errors.New("unexpected status code")
 
 func (r HttpRunner) Run(ctx context.Context, script Script) (*RunResponse, error) {
-	k6Timeout := time.Duration(script.Settings.Timeout) * time.Millisecond
+	if r.backoff == 0 {
+		panic("zero backoff, runner is misconfigured, refusing to DoS")
```

Comment:
If 0 is not an allowed value, isn't it better to use it as an indication that it should fall back to the default value?

Reply:
I'm not against changing this, but just for context: This value should always be the default, there is no exported way to override it. This check is here mostly as a safety measure for bugs, and also for package level unit tests that construct an `HttpRunner` directly. I initially chose to panic mostly to signal "your unit test is wrong".